tools/firtool/firtool.cpp

```diff
@@ -561,6 +568,11 @@ processBuffer(MLIRContext &context, TimingScope &ts, llvm::SourceMgr &sourceMgr,
   if (!disableOptimization) {
     pm.nest<firrtl::CircuitOp>().nest<firrtl::FModuleOp>().addPass(
         createSimpleCanonicalizerPass());
+    // TODO: Move this to the O1 pipeline.
```

Comment:
shouldn't this pass be the first pass after semantic checks are done so that other passes can take advantage of more freedom?